lib/modules/manager/dockerfile/extract.ts

```diff
@@ -96,36 +93,25 @@ function processDepForAutoReplace(
 }
 
 export function splitImageParts(currentFrom: string): PackageDependency {
+  let isVariable = false;
+  let cleanedCurrentFrom = currentFrom;
+
   // Check if we have a variable in format of "${VARIABLE:-<image>:<defaultVal>@<digest>}"
   // If so, remove everything except the image, defaultVal and digest.
-  let isVariable = false;
-  let cleanedCurrentFrom: string = currentFrom;
-  if (
-    currentFrom.startsWith(variableOpen) &&
-    currentFrom.endsWith(variableClose)
-  ) {
-    isVariable = true;
+  if (cleanedCurrentFrom?.includes(variableMarker)) {
+    const defaultValueRegex = regEx(/^\${.*?:-"?(?<value>.*?)"?}$/);
```

Comment:
Should this `*` be a `+`, otherwise the variable name can be empty?

```suggestion
    const defaultValueRegex = regEx(/^\${.+?:-"?(?<value>.*?)"?}$/);
```

Reply:
Good catch! Empty vars result in `syntax error: bad substitution`